src/app/pages/virtualization/components/all-instances/instance-details/instance-devices/instance-devices.component.ts

```diff
@@ -14,8 +25,43 @@ import { VirtualizationInstance } from 'app/interfaces/virtualization.interface'
     MatButton,
     MatCardModule,
     TranslateModule,
+    MatCardContent,
+    MatProgressSpinner,
+    TitleCasePipe,
   ],
 })
-export class InstanceDevicesComponent {
+export class InstanceDevicesComponent implements OnChanges {
   instance = input.required<VirtualizationInstance>();
+
+  devices = signal<VirtualizationDevice[]>([]);
+  isLoading = signal<boolean>(false);
+
+  constructor(
+    private ws: WebSocketService,
+    private errorHandler: ErrorHandlerService,
+  ) {}
+
+  ngOnChanges(changes: IxSimpleChanges<this>): void {
+    if (!(changes.instance.currentValue as unknown as VirtualizationInstance).id) {
+      return;
+    }
+
+    this.loadDevices();
+  }
+
+  loadDevices(): void {
+    this.isLoading.set(true);
+    this.ws.call('virt.instance.device_list', [this.instance()?.id])
```

Comment:
This check might send [undefined] instead of an empty array. Condition can be improved.

Reply:
It can't be `undefined`, we render only `selectedInstance()`, undefined won't render.

  <div
    ixDetailsHeight="rightside-content-hold"
    class="details-container"
    [class.details-container-mobile]="showMobileDetails()"
  >
    @if (selectedInstance()) {
      <ix-instance-details [instance]="selectedInstance()"></ix-instance-details>
    }
  </div>